src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -365,7 +365,7 @@ public Task<bool> HasChangesAsync(Solution solution, string? sourceFilePath, Can
             var editSession = _editSession;
             if (editSession == null)
             {
-                return Task.FromResult(false);
```

Comment:
>Task.FromResult(false); [](start = 23, length = 23)

I believe Boolean is optimized in Task.FromResult, we don't need ours.

Reply:
See https://github.com/dotnet/roslyn/pull/47117#discussion_r476756251. `Task.FromResult<TResult>` does not optimize for any cases.

Reply:
You may be thinking of `AsyncTaskMethodBuilder<TResult>`, which does optimize these cases for `async` methods that return synchronously.

Reply:
I see. You are right